go-controller/pkg/clustermanager/userdefinednetwork/controller.go

```diff
@@ -265,57 +370,46 @@ func updateCondition(conditions []metav1.Condition, cond *metav1.Condition) ([]m
 	return conditions, false
 }
 
-// UpdateSubsystemCondition may be used by other controllers handling UDN/NAD/network setup to report conditions that
-// may affect UDN functionality.
-// FieldManager should be unique for every subsystem.
-// If given network is not managed by a UDN, no condition will be reported and no error will be returned.
-// Events may be used to report additional information about the condition to avoid overloading the condition message.
-// When condition should not change, but new events should be reported, pass condition = nil.
-func (c *Controller) UpdateSubsystemCondition(networkName string, fieldManager string, condition *metav1.Condition,
-	events ...*util.EventDetails) error {
-	// try to find udn using network name
-	udnNamespace, udnName := template.ParseNetworkName(networkName)
-	if udnName == "" {
-		return nil
-	}
-	udn, err := c.udnLister.UserDefinedNetworks(udnNamespace).Get(udnName)
-	if err != nil {
-		return nil
-	}
+func (c *Controller) cudnNeedUpdate(_ *userdefinednetworkv1.ClusterUserDefinedNetwork, _ *userdefinednetworkv1.ClusterUserDefinedNetwork) bool {
```

Comment:
well there may be changes that will trigger the update logic that we could filter out here. If there were like labels added to the CUDN, or maybe the status was updated of the CUDN. It would be worth thinking through what we actually care about changing here, and filtering for only that.